duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -313,6 +321,27 @@ private String canonQueryString()
     return Util.join(args.toArray(), "&");
   }
 
+  private String canonXDuoHeaders(){
+    ArrayList<String> canonList = new ArrayList<String>();
+    for (String name : additionalHeaders.keySet()){
+      String value = additionalHeaders.get(name);
+      canonList.add(name + value);
+    }
+    return Util.join(canonList.toArray(), String.valueOf(Character.MIN_VALUE));
+  }
+
+  private void validateAdditionalHeader(String name, String value) throws IllegalArgumentException{
+    if (name == null || name.length() == 0){
+      throw new IllegalArgumentException("Not allowed 'Null' or empty header name");
+    } else if (value == null || value.length() == 0){
+      throw new IllegalArgumentException("Not allowed 'Null' or empty header value");
+    } else if (!name.toLowerCase().startsWith("x-duo-")){
+      throw new IllegalArgumentException("Additional headers must start with \'X-Duo-\'");
```

Comment:
I'm not sure we want to only allow x-duo headers here.  we definitely should only use the x-duo headers in the c14n, but we might want to allow others to be sent.